### ISSUE TITLE
Bug in avg time calculation

### ISSUE THREAD

**Issue by Raghwendra-Dey** [Raghwendra-Dey/productivity_meter#101]
After #86 we can remove a problem, and thats taken time would be removed but theres some miscalculation due to this.
Follow these steps for reproducing the bug: (wait for suitable time according to your approximations)
* do `+1` to get avg of around 2 secs
* then do `+1` to make the avg 3-4 secs
* then do `+1` to make the avg 4-5 secs
* then remove do a `-1`, you should get avg to be 3-4 secs(prev value)
* then again do `-1` untill you get avg 0.
* then do a `+1` immediately after you have done the last `-1` which made the avg 0.
* you would see a much larger avg than expected. Since you have done the last `+1` immediately, it should have shown you a much smaller value(like maybe 0/1).

I think this is something to do with accumulation of all the previous times(I mean all the `+1`'s you did before all those times are getting added up). Maybe the starting time used during finding the time difference is not getting updated or maybe theres some variable which is also getting restored to the previous value which shouldn't have been.

**Comment by Raghwendra-Dey** [comment]
@Bbiswabasu you have worked on the deque thing. would you like to take this up?

**Comment by Bbiswabasu** [comment]
Suppose record is clicked at t=0, 3rd task was finished at t=3s and I add 4th task at t=4s. Avg shown will be 1s. If I press -1 and again press +1 at t=8s, avg shown ( According to my implementation) is 8/4=2s irrespective of when -1 was clicked. Is this part fine?

**Comment by rsriramc** [comment]
I think this is an ambiguous thing, since in one point of view we can think that the whole window can be thought of used for one problem since the problems that were minused/removed were not completed or done. So it means that the whole time is now used for this one. But thats not the case as per the average time scenario. Since it refers to the case that only one thing is done at a time.

**Comment by Bbiswabasu** [comment]
So should I reset starting time for a problem when -1 is clicked?

**Comment by Raghwendra-Dey** [comment]
> Suppose record is clicked at t=0, 3rd task was finished at t=3s and I add 4th task at t=4s. Avg shown will be 1s. If I press -1 and again press +1 at t=8s, avg shown ( According to my implementation) is 8/4=2s irrespective of when -1 was clicked. Is this part fine?

No, but if you think this from the user's perspective, the -1 button is just there that lets say I solved a problem, and I did a +1 for that, but later I realised that the soln is not correct so I would do a -1(which means I don't want to take into account whatever time is gone in the last problem. For sure, its not correct to include that amount of time to the next problem, since I haven't utilised it for the next problem, instead it was wasted. 
So, I think we should only count the time for the next button not the whole time accumulated(even the once which I poped out)
Let me know, If I am wrong.
